### PR TITLE
K8s support

### PR DESCRIPTION
**Pull request by comzyh**

### CODE REVIEW

File: cmd/master/main.go [hunk 1]
```diff
@@ -1,21 +1,93 @@
...
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"os"
 	"strings"
 
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
```
Comment (Lcch) [inline cmd/master/main.go]: third-party packages should be at the bottom of others imports

File: cmd/reducer/main.go [hunk 2]
```diff
@@ -4,8 +4,8 @@ import (
 	"flag"
 	"os"
 
-	"github.com/ceph/go-ceph/rados"
-	"github.com/naturali/NiServer/pkg/log"
+	records "github.com/naturali/kmr/records/adaptor"
+	"github.com/naturali/kmr/util/log"
```
Comment (Lcch) [inline cmd/reducer/main.go]: imports order: third party should be at last

File: master/k8sutil.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+		ReplicaSets(master.namespace).Create(replicaSet)
+}
+
+func (master *Master) startWorker(phase string) error {
+	var rs v1beta1.ReplicaSet
+	switch phase {
+	case mapPhase:
+		rs = master.generateReplicaSet(master.JobDesc.Map.commad, master.JobDesc.Map.Image, int32(master.JobDesc.Map.NWorker))
+	case reducePhase:
+		rs = master.generateReplicaSet(master.JobDesc.Reduce.commad, master.JobDesc.Map.Image, int32(master.JobDesc.Map.NWorker))
```
Comment (Lcch) [inline master/k8sutil.go]: master.JobDesc.Map.Image -> master.JobDesc.Reduce.Image
master.JobDesc.Map.NWorker -> master.JobDesc.Reduce.NWorke

File: master/master.go [hunk 4]
```diff
@@ -1,14 +1,17 @@
...
 
 import (
-	"golang.org/x/net/context"
+	"fmt"
 	"math/rand"
 	"net"
 	"sync"
 	"time"
 
+	"golang.org/x/net/context"
```
Comment (Lcch) [inline master/master.go]: import style

File: master/k8sutil.go [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/client-go/pkg/api/v1"
+	v1beta1 "k8s.io/client-go/pkg/apis/extensions/v1beta1"
+)
+
+func (master *Master) generateReplicaSet(name string, command []string, image string, replicas int32) v1beta1.ReplicaSet {
```
Comment (xiang90) [inline master/k8sutil.go]: createReplicaSet

Comment (xiang90) [inline master/k8sutil.go]: oh. this is the create the struct pointer. so NewReplicaSet

**Pull request merged**